src/routes/login.tsx

```diff
@@ -49,6 +58,9 @@ export function LoginForm() {
     try {
       const client = await createClient(role, key);
       setSpvWalletClient(client);
+
+      login(client);
+
```

Comment:
Why the additional `login` function call is necessary?
Cannot we rely on the fact that the `client` is not empty? As we did before your changes?

Reply:
To split authentication logic. This function set isAdmin and set isAuthenticated. Using these values helps to separate access to pages and functionality.

Reply:
> Cannot we rely on the fact that the client is not empty? As we did before your changes?

So I'm still confused why we cannot "compute" these `isAdmin & isAuthenicated` flags like we did previously:
```
const isAuthenticated = !!spvWalletClient;
  const isAdmin = isAuthenticated && spvWalletClient?.role === Role.Admin;
  const contextValue = {
    isAuthenticated,
    isAdmin,
  };
```